### DISCUSSION TITLE
Flashing and Using

### DISCUSSION THREAD

**Discussion by versionDefect** [awawa-dev/HyperSerialESP32#45] [Q&A]
Hi all, 

TLDR: redid my ambilight setup to make it cleaner and decided to user a new microcontroller along with it. 

I cannot for the life of me get this to work. I've got two different ESP32's AND ESP8266's and none of them are cooperating with me. Is there any way after i flash i can confirm the flash was successful? 

`2023-11-19T10:32:42.610Z [LEDDEVICE_ADALIGHT] Could not detect HyperSerialEsp8266/HyperSerialESP32 device
2023-11-19T10:32:49.250Z [LEDDEVICE_ADALIGHT] Device disabled, device 'adalight' signals error: 'Timeout writing data to ttyAMA0'
`

I get that on every device. I promise I'm usually not this stupid lol.

**Comment by awawa-dev** [comment]
Hi
Post full logs (here or preferable on pastebin) taken after 2-3 minutes after starting HyperHDR

**Reply by versionDefect** [reply]
https://pastebin.com/mKSN5PCk

Sorry I've really racked my brain out at this. z

EDIT: I know in the log it says display device is unavailable, I've brought the pi to my desk for debugging so there is no "input" but its running an effect.

**Reply by awawa-dev** [reply]
It does not matter. The operating system serial driver is damaged or you are not using the appropriate serial device, e.g. instead of USB some other RPi internal device like Bluetooth:
```
2023-11-20T05:55:30.683Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:221) portName:          ttyAMA0
2023-11-20T05:55:30.683Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:222) systemLocation:    /dev/ttyAMA0
2023-11-20T05:55:30.684Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:223) description:       
2023-11-20T05:55:30.684Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:224) manufacturer:      
2023-11-20T05:55:30.684Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:225) productIdentifier: 0x0
2023-11-20T05:55:30.684Z [LEDDEVICE_ADALIGHT] (ProviderRs232.cpp:226) vendorIdentifier:  0x0
```
vendorIdentifier and productIdentifier are not valid: 0 and that can't work. What's the OS/kernel and RPi version? Please attach dmesg logs.

**Reply by versionDefect** [reply]
Attached is the DMESG logs. 
Edit: didn't mean to hit send yet lol. ~~https://pastebin.com/f1Ldctce~~. let me get the os and kernal info. 

Another edit: I'm trying a different controller on that log. let me go back to the first controller to keep consistent logs.

**Reply by versionDefect** [reply]
![image](https://github.com/awawa-dev/HyperSerialESP32/assets/37282503/fda0e28d-8c47-4016-9223-0e433c7f1dee)

Hyper HDR Log: https://pastebin.com/uGe7Svz9 
DMESG Log: https://pastebin.com/Jh9KUBh0

Those logs are with this controller: https://www.amazon.com/gp/product/B0B291LZ99/ref=ppx_yo_dt_b_asin_title_o09_s00?ie=UTF8&psc=1

**Reply by awawa-dev** [reply]
PL011 at AMA0 in dmesg is for 99% RPi Bluetooth device, not ESP

**Reply by versionDefect** [reply]
thats.... interesting lol. Let me get you some logs with the other device I also have. One minute

**Reply by versionDefect** [reply]
![image](https://github.com/awawa-dev/HyperSerialESP32/assets/37282503/d29a0c62-8ed5-46e2-8994-d5e826bdc35a)
Apologies for the wait, needed to lend a hand to a friend quickly
New ESP
Hyper HDR Log https://pastebin.com/efz9sRwi 
DMESG logs: https://pastebin.com/MQ1EA1Sj
https://www.aliexpress.com/item/3256802529239587.html?spm=a2g0o.order_list.order_list_main.5.7ff918020x8iW0

**Reply by awawa-dev** [reply]
This time proper device is selected (idvendor=1a86 idproduct=55d4) HyperHDR could not reset it at beginning to confirm HyperSerial firmware but it doesn't mean it doesn't work because there is no error. Not every ESP reacts the same way for reset command (it needs proper hardware be present on the board, behavior controlled by UART), for some models it simply won't work and I don't know the board you are using. Didn't have you problems with the firmware flashing or did the board  require to be put in flash mode using board buttons (flasher software reset didn't work)?

**Reply by versionDefect** [reply]
> This time proper device is selected (idvendor=1a86 idproduct=55d4) HyperHDR could not reset it at beginning to confirm HyperSerial firmware but it doesn't mean it doesn't work because there is no error. Not every ESP reacts the same way for reset command (it needs proper hardware be present on the board, behavior controlled by UART), for some models it simply won't work and I don't know the board you are using. Didn't have you problems with the firmware flashing or did the board  require to be put in flash mode using board buttons (flasher software reset didn't work)?

Yes for one of them i have the S2 iirc i had to hold down the 0 button and tap reset once to get it into DFU mode. The other one i think i just had to hit a button to get it to flash but i was under the impression once its flashed, then when it resets / loses power. it runs off the flash? Am i wrong for thinking that? (genuine question here haha)

**Reply by awawa-dev** [reply]
Since there are 2 physical buttons on the board, the manufacturer may have abandoned the option of software resetting the system. But this is very inconvenient: firstly when uploading any firmware, and secondly, HyperHDR cannot reset the board to restore it to its original state after the firmware startup. It is better to avoid such models.
Try to read the number of rendered frames by performing this test: https://www.hyperhdr.eu/2023/02/ultimate-guide-on-how-to-build-led.html#chapter9. Although there is no guarantee that communication from the other side (from ESP to HyperHDR) works without a reset.

**Reply by versionDefect** [reply]
> Since there are 2 physical buttons on the board, the manufacturer may have abandoned the option of software resetting the system. But this is very inconvenient: firstly when uploading any firmware, and secondly, HyperHDR cannot reset the board to restore it to its original state after the firmware startup. It is better to avoid such models.
> 
> Try to read the number of rendered frames by performing this test: https://www.hyperhdr.eu/2023/02/ultimate-guide-on-how-to-build-led.html#chapter9. Although there is no guarantee that communication from the other side (from ESP to HyperHDR) works without a reset.

I see. I've been using 8266's up till now haha (but the older chip with 1mb speeds iirc). I'll grab a newer ESP32 and will keep you updated. thanks again! :-)

**Reply by versionDefect** [reply]
Hey again, Since that post I bought the exact one thats in this repo just to face the exact same issue. 

![image](https://github.com/awawa-dev/HyperSerialESP32/assets/37282503/71e236b6-2011-459c-b932-708ae80f2f38)

HyperHDR Logs. https://pastebin.com/jqTrCBCM

adding to this i was getting really bizzare logs from the ESPHome Flasher after it flashed. 

https://pastebin.com/FnAAvTwe

what do you think could be going wrong here because I'm at a total loss lol.

**Reply by awawa-dev** [reply]
Maybe the bootloader is missing/damaged. Upload the recovery(factory) version firmware at proper address.

**Reply by versionDefect** [reply]
dumb question but where do you think I could find that? I can't seen to find anything mentioning anything on the sellers website. https://www.aliexpress.us/item/2251832629802644.html?spm=a2g0o.order_list.order_list_main.5.22f91802ACZrAw&gatewayAdapt=glo2usa

**Reply by awawa-dev** [reply]
zip archive in the release section, general instructions on the main page. If flashing has worked for you so far, you probably won't have to manually put the board into DFU mode. The CP2104 is not the same chip that is typically used in this board model (CH9102X). Although it should work, but it does cast some doubt on the version of this board.
